sentry/src/main/java/io/sentry/FullyDisplayedReporter.java

```diff
@@ -0,0 +1,42 @@
+package io.sentry;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class FullyDisplayedReporter {
+
+  private static final @NotNull FullyDisplayedReporter instance = new FullyDisplayedReporter();
+
+  private final @NotNull List<FullyDrawnReporterListener> listeners = new ArrayList<>();
+
+  private FullyDisplayedReporter() {}
+
+  public static @NotNull FullyDisplayedReporter getInstance() {
+    return instance;
+  }
+
+  public void registerFullyDrawnListener(final @NotNull FullyDrawnReporterListener listener) {
+    synchronized (listeners) {
```

Comment:
`m`: I think we could use `CopyOnWriteArrayList` and would not need the `synchronized` block. I think it's also cheaper to iterate over the concurrent collection, rather than locking it.